gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/ProductDependencyIntrospectionPlugin.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableMap;
+import groovy.lang.Closure;
+import java.io.File;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.util.GFileUtils;
+
+public final class ProductDependencyIntrospectionPlugin implements Plugin<Project> {
+    private static final Logger log = Logging.getLogger(ProductDependencyIntrospectionPlugin.class);
+    private static final String PRODUCT_DEPENDENCIES_CONFIGURATION = "productDependencies";
+
+    @Override
+    public void apply(Project project) {
+        createGetMinimumProductVersion(project);
+
+        project.getConfigurations().create(PRODUCT_DEPENDENCIES_CONFIGURATION, conf -> {
+            conf.setCanBeResolved(false);
+            conf.setDescription("Exposes minimum, maximum versions of product dependencies as constraints");
+
+            Optional<List<ProductDependency>> allProductDependencies = getAllProductDependencies(project);
+            allProductDependencies.ifPresent(productDependencies -> conf
+                    .getDependencies()
+                    .addAll(createAllProductDependencies(project, productDependencies)));
+
+            if (!allProductDependencies.isPresent()) {
+                log.info("Lock file not present, not populating product dependencies configuration: {}", conf);
+            }
+        });
+    }
+
+    private static void createGetMinimumProductVersion(Project project) {
+        project
```

Comment:
could we not use the configuration to get the product version?

Reply:
this is just easier since it's what drives the configuration anyway